### PR TITLE
Add `/views/transaction-confirmation` endpoint

### PR DESCRIPTION
**Pull request by fmrsabino**
- Adds a `POST /v1/chains/:chainId/safes/:safeAddress/views/transaction-confirmation`
- The main goal of this endpoint is to return all the data required by the frontend apps to render the Transaction Confirmation Screen.
- Clients are expected to migrate to this endpoint from the `POST /v1/chains/:chainId/data-decoder` endpoint as this endpoint would be extended to provide more data to the Transaction Confirmation Screen.
- The Data Decoder endpoint would still be available as-is.
- This endpoint is currently experimental. To enable it, set `FF_CONFIRMATION_VIEW` to `true`.

### Future Work
Due to the experimental stage of the endpoint and its development cycle, tests will be added once we have a more stable to schema to test the endpoint against with.

### CODE REVIEW

File: src/routes/transactions/transactions-view.controller.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+} from '@/routes/common/entities/transaction-data.dto.entity';
+import { TransactionsViewService } from '@/routes/transactions/transactions-view.service';
+import { DataDecodedRepositoryModule } from '@/domain/data-decoder/data-decoded.repository.interface';
+
+@ApiTags('transactions')
+@Controller({
+  path: '',
+  version: '1',
+})
+export class TransactionsViewController {
```
Comment (iamacook) [inline src/routes/transactions/transactions-view.controller.ts:24]: We should keep in mind to generify this if we move towards more `views` endpoints.

File: src/routes/transactions/transactions-view.controller.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  @HttpCode(200)
+  @ApiOkResponse({ type: ConfirmationView })
+  @Post('chains/:chainId/safes/:safeAddress/views/transaction-confirmation')
+  @ApiOperation({
+    summary: 'Confirm Transaction View',
+    description: 'This endpoint is experimental and may change.',
+  })
+  async getTransactionConfirmationView(
+    @Param('chainId') chainId: string,
+    @Param('safeAddress') safeAddress: string,
```
[the remark this answers is not in the document]
Comment (fmrsabino) [reply src/routes/transactions/transactions-view.controller.ts]: Added in [deb95e9](https://github.com/safe-global/safe-client-gateway/pull/1419/commits/deb95e9f30258d36556a12481e3bba7f722fff1f)

File: src/routes/transactions/transactions-view.controller.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  @ApiOkResponse({ type: ConfirmationView })
+  @Post('chains/:chainId/safes/:safeAddress/views/transaction-confirmation')
+  @ApiOperation({
+    summary: 'Confirm Transaction View',
+    description: 'This endpoint is experimental and may change.',
+  })
+  async getTransactionConfirmationView(
+    @Param('chainId', new ValidationPipe(NumericStringSchema)) chainId: string,
+    @Param('safeAddress', new ValidationPipe(AddressSchema))
+    safeAddress: string,
```
Comment (iamacook) [inline src/routes/transactions/transactions-view.controller.ts:39]: Nit: as we're using the schema, we can make this stricter as well.
```suggestion
    safeAddress:`0x${string}`,
```

Comment (fmrsabino) [reply src/routes/transactions/transactions-view.controller.ts]: Done in [cde751e](https://github.com/safe-global/safe-client-gateway/pull/1419/commits/cde751ed3153d9f53cc274ebf631efc0e0ac7863)

**Review by iamacook — APPROVED** [review]

**Pull request merged**